java/src/IceGridGUI/src/main/java/com/zeroc/IceGridGUI/Coordinator.java

```diff
@@ -1250,419 +1252,375 @@ enum TrustDecision { YesAlways, YesThisTime, No };
                         {
-                            break;
+                            try
+                            {
+                                SwingUtilities.invokeAndWait(() ->
+                                                            {
+                                                                JOptionPane.showMessageDialog(parent, ex.toString(), "Error loading keystore",
+                                                                                            JOptionPane.ERROR_MESSAGE);
+                                                            });
+                                break;
+                            }
+                            catch(java.lang.InterruptedException e)
+                            {
+                            }
+                            catch(java.lang.reflect.InvocationTargetException e)
+                            {
+                                break;
+                            }
                         }
+                        throw certificateException;
                     }
-                    return false;
-                }
-
-                boolean validDate = true;
-                boolean trustedCA = false;
-                boolean validAlternateName = false;
 
-                //
-                // Check the certificate date is valid.
-                //
+                    TrustDecision decision = new AcceptInvalidCertDialog().show(serverCertificate);
 
-                java.util.Date now = new java.util.Date();
-                if(now.getTime() > cert.getNotAfter().getTime() || now.getTime() < cert.getNotBefore().getTime())
-                {
-                    validDate = false;
-                }
-
-                String remoteAddress = null;
-                for(com.zeroc.Ice.ConnectionInfo p = info.underlying; p != null; p = p.underlying)
-                {
-                    if(p instanceof com.zeroc.Ice.IPConnectionInfo)
+                    if(decision == TrustDecision.YesThisTime)
                     {
-                        remoteAddress = ((com.zeroc.Ice.IPConnectionInfo)p).remoteAddress;
-                        break;
+                        _transientCert = serverCertificate;
+                        System.out.println("Trust this time");
+                        return;
                     }
-                }
-
-                //
-                // Check server alternate names match the connection remote address
-                //
-                if(remoteAddress != null)
-                {
-                    try
+                    else if(decision == TrustDecision.YesAlways)
                     {
-                        Collection<java.util.List<?>> altNames = cert.getSubjectAlternativeNames();
-                        if(altNames != null)
+                        try
                         {
-                            for(java.util.List<?> l : altNames)
+                            String CN = "";
```

Comment:
Why switch to common name here? Pretty sure subjectAlt name is still the preferred field.